### PR TITLE
Gjoranv/vagrant gui switch

### PR DESCRIPTION
**Pull request by gjoranv**
Please note the `<TODO>`s for the environment vars to specify box and box url with GUI. Feel free to commit to this branch or specify what the section should say in a comment.

FYI: @bjorncs

### CODE REVIEW

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+disable_gui = ENV['VESPA_DISABLE_VAGRANT_GUI']
```
Comment (geirst) [inline vagrant/Vagrantfile]: Consider rename to VESPA_VAGRANT_DISABLE_GUI to use the same prefix as the other variables

File: vagrant/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <!-- Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
 
-# Create C++ dev environment on CentOS using VirtualBox and Vagrant
+# Create C++ or Java dev environment on CentOS using VirtualBox and Vagrant
```
Comment (geirst) [inline vagrant/README.md]: Consider rename to "C++ / Java" or "C++ and/or Java"

Comment (geirst) [reply vagrant/README.md]: Also update the link name in the top-level readme file.

File: vagrant/README.md [hunk 3]
```diff
@@ -12,16 +12,28 @@
...
+
+##### a. For a dev environment with plain centos/7 and no GUI:
+
+    export VESPA_VAGRANT_VM_BOX=centos/7
+    export VESPA_DISABLE_VAGRANT_GUI=true
+
+##### b. For a dev environment with GUI and CLion:
+
+    export VESPA_VAGRANT_VM_BOX=<TODO>
+    export VESPA_VAGRANT_VM_BOX_URL=<TODO>
```
Comment (geirst) [inline vagrant/README.md]: Sub-section for building the box would contain something like this:

Create centos7-desktop box

1) Install packer
Follow guide at https://www.packer.io/intro/getting-started/install.html.

2) Clone boxcutter centos repo
git clone https://github.com/boxcutter/centos.git

3) Build box
./bin/box build centos7-desktop.json virtualbox

Example exports:
export VESPA_VAGRANT_VM_BOX="centos7-desktop"
export VESPA_VAGRANT_VM_BOX_URL="/Users/foo/git/boxcutter/centos/box/virtualbox/centos7-desktop-xx.yyyy.z.box"

**Review by geirst — COMMENTED** [review]
Overall looks good, please have a look at my comments

**Review by vekterli — APPROVED** [review]
:+1: See @geirst 's comments

**Comment by gjoranv** [comment]
@geirst Updated according to your comments. PTAL

**Review by geirst — APPROVED** [review]
Very good 👍

**Pull request merged**